packages/compass-crud/src/components/insert-document-dialog.jsx

```diff
@@ -224,19 +229,43 @@ class InsertDocumentDialog extends React.PureComponent {
         data-testid="insert-document-modal"
         minBodyHeight={spacing[6] * 2} // make sure there is enough space for the menu
       >
-        <div className="insert-document-views">
-          <ViewSwitcher
-            dataTestId="insert-document-dialog-view"
+        <div className={toolbarStyles}>
+          <SegmentedControl
             label="View"
-            buttonLabels={['JSON', 'List']}
-            showLabels={false}
-            iconClassNames={['curly-bracket', 'fa fa-list-ul']}
-            activeButton={currentView}
-            disabled={this.hasErrors()}
-            onClick={this.switchInsertDocumentView.bind(this)}
-          />
+            size="small"
+            value={currentView}
+            aria-controls={documentViewId}
+            onChange={this.switchInsertDocumentView.bind(this)}
+          >
+            <SegmentedControlOption
+              disabled={this.hasErrors()}
+              data-testid="insert-document-dialog-view-json"
+              aria-label="E-JSON View"
+              value="JSON"
+              onClick={(evt) => {
+                // The value changing occurs in the `onChange` in the `SegmentedControl`.
+                evt.preventDefault();
```

Comment:
I get not handling change event here, but why do we need to prevent default?

Reply:
Probably because SegmentedControlOption is a submit button? https://github.com/mongodb/leafygreen-ui/blob/main/packages/segmented-control/src/SegmentedControlOption.tsx#L348-L362

Reply:
Aha, probably, not obvious from the comment, so maybe would be good to clarify for the future reference. Seems like you can't even pass custom props to the button... Have we reported the issue to leafygreen?

Reply:
Updated the comment to describe why we're doing that and asked in their channel: https://mongodb.slack.com/archives/CFFA5BS79/p1667312599633259
👍